packages/owlbot-bootstrapper/common-container/test/github-authenticator.test.ts

```diff
@@ -40,12 +41,15 @@ secretManagerClientStub.accessSecretVersion.resolves([
   },
 ]);
 
+const signStub = sinon.stub(jwt, 'sign');
+
 describe('behavior of Github Authenticator Class', async () => {
   it('should create the right type of object', async () => {
     const githubAuthenticator = new GithubAuthenticator(
       'projectId',
       '2345567',
-      secretManagerClientStub
+      secretManagerClientStub,
+      signStub as unknown as typeof jwt.sign
```

Comment:
Generally we don't pass method stubs around as parameters (you would pass a fake or a mock).

If using stubs, you would normally do:

```
const signStub = sinon.stub(jwt, 'sign').returns(something); // or .resolves(something);
```

This would stub out the response of jwt's sign method when it's called. As this is written, we're passing this sinon stub method into this method to be called which is somewhat odd.

Reply:
Hm, I don't see why this would be a problem, since it's not changing the tests' functionality. In the original test, we're passing a fake private key, and creating a jwt and then making a github call with that jwt. Ultimately though, this test was just making sure the github call was made; it was not testing that the jwt was created correctly. This test just makes it so that we're not testing the jwt dependency directly, but we don't need to be testing the output of what the sign function does.